### PR TITLE
Vip 39 five two processing

### PR DESCRIPTION
**Pull request by tank157**
[VIP-39](https://democracyworks.atlassian.net/browse/VIP-39?atlOrigin=eyJpIjoiODNkMTMxNjNiYTFjNGE1YmFhOGM3NDIxN2FmMjBjZmIiLCJwIjoiaiJ9)

There's a lot in here:
* Finally incorporating that big pipeline refactor from a year ago. The actual work for the new 5.2 elements starts on July 15, 2020.
* VIP-39, VIP-40, VIP-41, VIP-42, VIP-43, and VIP-44

Most of the work for each of the individual cards are in commits named with the relevant card name, so that should help somewhat.

I've run a lot of tests, but there's so much in here I may have missed some.

### CODE REVIEW

**Comment by lizstarin** [comment]
_scans headings_ `Files changed 223` 😱

**Comment by tank157** [comment]
Yeah, this one may be easier to look at just by the commits that start on July 15th-ish, the first one with `VIP-40` in the name and the ones after. I mean, feel free to look at any of them, but before that is mostly merging in the old pipelines refactor work and adjusting a few things to some data-processor changes that happened recently.

File: src/vip/data_processor/validation/v5/street_segment.clj [hunk 1]
```diff
@@ -40,6 +40,130 @@
                           :fatal
                           :end-house-number))
 
+(defn valid-house-number-fix-or-nil?
+  "Validate the prefix or suffix house number. They are restricted to
+   alphabetical characters, numbers, and the '/' character."
+  [house-number-fix]
+  (or (str/blank? house-number-fix)
+      (not (str/blank? (re-find #"^[a-zA-z0-9\/-]+$" house-number-fix)))))
```
Comment (lizstarin) [inline src/vip/data_processor/validation/v5/street_segment.clj:48]: Hyphens too?

File: test/vip/data_processor/validation/v5/street_segment_test.clj [hunk 2]
```diff
@@ -258,3 +258,147 @@
...
+(deftest ^:postgres validate-house-number-prefix-no-includes-all-streets
+  (let [errors-chan (a/chan 100)
+        ctx {:xml-source-file-path (xml-input "v5-street-segment-house-number-prefix.xml")
+             :errors-chan errors-chan}
+        out-ctx (-> ctx
+                    psql/start-run
+                    xml/load-xml-ltree
+                    v5.ss/validate-no-includes-all-streets-with-house-number-prefix)
+        errors (all-errors errors-chan)]
+    (testing "cannot includes all addresses with house number prefix"
```
Comment (lizstarin) [inline test/vip/data_processor/validation/v5/street_segment_test.clj:295]: You mean _streets_, not _addresses_ here, right?

File: src/vip/data_processor/db/postgres.clj [hunk 3]
```diff
@@ -303,10 +303,11 @@
 
 (defn delete-from-xml-tree-values
   [{:keys [import-id] :as ctx}]
-  (log/info "Dropping from xml_tree_values")
-  (korma/exec-raw
-   (:conn xml-tree-values)
-   ["delete from xml_tree_values where results_id = ?" [import-id]])
+  (when-not (:keep-feed-on-complete? ctx)
+    (log/info "Dropping from xml_tree_values")
+    (korma/exec-raw
```
[the remark this answers is not in the document]
Comment (tank157) [reply src/vip/data_processor/db/postgres.clj:307]: Yes (in fiver), but probably no in this repo, assuming we do get to pick up work again in fiver post election. If for some reason we need to just carry on with data-processor, then that could be a consideration.

File: src/vip/data_processor/pipelines.clj [hunk 4]
```diff
@@ -0,0 +1,27 @@
+(ns vip.data-processor.pipelines
+  (:require [vip.data-processor.pipelines.csv.v3 :as csv.v3]
+            [vip.data-processor.pipelines.csv.v5 :as csv.v5]
+            [vip.data-processor.pipelines.xml.v3 :as xml.v3]
+            [vip.data-processor.pipelines.xml.v5 :as xml.v5]))
+
+(defn choose-pipeline [ctx]
```
[the remark this answers is not in the document]
Comment (tank157) [reply src/vip/data_processor/pipelines.clj:7]: I am not at this time, I'll add a test for it.

File: dev-src/dev/core.clj [hunk 5]
```diff
@@ -29,7 +29,7 @@
   (let [file (if (zip/xml-file? filename)
                (java.nio.file.Paths/get filename (into-array [""]))
                (java.io.File. filename))
-        result (pipeline/process pipeline file)]
+        result (time (pipeline/process pipeline file))]
```
[the remark this answers is not in the document]
Comment (tank157) [reply dev-src/dev/core.clj]: We don't for the `dev-src` runs, I think it just gets printed out to the console.

File: src/vip/data_processor/validation/zip.clj [hunk 6]
```diff
@@ -22,18 +22,41 @@
...
+  ([ctx]
+   (assert-max-zip-size ctx (config [:max-zipfile-size] 3221225472)))
+  ([ctx max-zipfile-size]
+   (let [file-path (:file ctx)]
+     (cond
+       (zip-file? file-path)
+       (let [size (get-uncompressed-size file-path)]
+         (if (> size (Long. max-zipfile-size))
+           (assoc ctx :stop (too-big-msg size max-zipfile-size))
+           ctx))
```
[the remark this answers is not in the document]
Comment (tank157) [reply src/vip/data_processor/validation/zip.clj]: Yeah, the value that gets assoc'd to the `:stop` keyword is the messaging, and I'm pretty sure that gets communicated out somewhere when processing is stopped.

File: src/vip/data_processor/validation/v5/street_segment.clj (hunk 1 above)
Comment (tank157) [reply src/vip/data_processor/validation/v5/street_segment.clj:48]: Good eye, yes, I did include the hyphen as a valid character. I'm not really recalling why at the moment though, I went to look at the spec and it says letters (aka alphabetical letters and or numerical digits) or slashes, but not hyphens.

Ah, found it, a comment from Franklin in Slack:
```
And I think a ‘-’ might also be supplied, which I think should be a valid character.
```

File: test/vip/data_processor/validation/v5/street_segment_test.clj (hunk 2 above)
Comment (tank157) [reply test/vip/data_processor/validation/v5/street_segment_test.clj:295]: Yup, good catch. Some leftover copypasta I think.

**Pull request merged**